src/execution.rs

```diff
@@ -58,13 +58,24 @@ impl File {
         }
     }
 
-    pub(self) fn check_globals(&self, globals: &Globals) -> Result<(), ExecutionError> {
+    pub(self) fn check_globals(&self, globals: &mut Globals) -> Result<(), ExecutionError> {
         for global in &self.globals {
             match globals.get(&global.name) {
                 None => {
-                    return Err(ExecutionError::MissingGlobalVariable(
-                        global.name.as_str().to_string(),
-                    ));
+                    if let Some(default) = &global.default {
+                        globals
+                            .add(global.name.clone(), default.to_string().into())
+                            .map_err(|_| {
+                                ExecutionError::DuplicateVariable(format!(
+                                    "global variable {} already defined",
+                                    global.name
+                                ))
+                            })?;
```

Comment:
This could maybe be a panic, or an `unreachable!()` inside the `map_err` callback, since this code only gets executed when `globals.get` fails.  (Though on the other hand, this is safer, and I can't imagine the redundant check adds enough overhead to insist on removing it.)